src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | `circle` | `rectangle`;
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Figure['shape'] = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sorted: number[] = [a, b, c].sort((el1, el2) => el2 - el1);
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('all sides should be positive number');
+    }
+
+    if (sorted[0] >= sorted[1] + sorted[2]) {
+      throw new Error(
+        'the longest side of a triangle must be  >= than a sum of two others',
+      );
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const semiparam = (this.a + this.b + this.c) / 2;
```

Comment:
`semiparam` sounds bit weird :)

I would suggest `halfSum` or something like it